js/https.js

```diff
@@ -0,0 +1,171 @@
+let knownMixedContentList
+load.JSONfromLocalFile(settings.getSetting('httpsWhitelist'), (wl) => knownMixedContentList = wl)
+
+class HTTPS {
+
+    constructor () {
+        this.isReady = false
+        this.db = null
+        this.dbObjectStore = 'https'
+        db.ready().then(() => { 
+          this.isReady = true
+          this.db = db 
+        })
+
+        return this
+    }  
+
+    pipeRequestUrl (reqUrl, tab) {
+        return new Promise((resolve) => {
+            if (!this.isReady) {
+                console.warn('HTTPS: .pipeRequestUrl() this.db is not ready')
+                return resolve(reqUrl)
+            }
+
+            reqUrl = reqUrl.toLowerCase()
+
+            // Only deal with http calls
+            const protocol = URLParser.extractProtocol(reqUrl).protocol
+            if (!protocol.indexOf('http:') === 0) return resolve(reqUrl)
+
+            // Obey global settings (options page)
+            if (!settings.getSetting('httpsEverywhereEnabled')) return resolve (reqUrl)
+
+            // Check bundled https whitelist of known mixed content sites
+            if (knownMixedContentList && knownMixedContentList[tab.site.domain]) {
+                console.log(`HTTPS: ${tab.site.domain} has known mixed content. skip upgrade check.`)  
```

Comment:
I think we still want to check any request that doesn't match a whitelisted domain. This skips all requests for a site that's in the whitelist?

Reply:
Yes! Nice catch, thanks. Added an `if (isMainFrame)` check before this rule so we are only skipping upgrade checks on main_frame requests now rather than all requests.